syft/src/main/java/org/openmined/syft/data/Dataset.kt

```diff
@@ -0,0 +1,23 @@
+package org.openmined.syft.data
+
+
+interface Dataset {
+    /**
+     * An abstract class representing a :class:`Dataset`.
+     * All datasets that represent a map from keys to data samples should subclass
+     * it. All subclasses should overwrite :method:`getItem`, supporting fetching a
+     * data sample for a given key. Subclasses could also optionally overwrite
+     * :meth:`len`.
+     */
+
+    /**
+     * This method is called to return the size of the dataset, needs to be overridden.
+     */
+    fun length() : Int
+
+    /**
+     * This method is called to fetch a data sample for a given key.
+     */
+    fun getItem(index: Int) : Pair<KTensor, KTensor>
```

Comment:
KTensor (SyftTensor) is just a util class to help create IValue later, and keep the data and the shape separated

Reply:
Let's leave it like this by the time being.
Could you please open an issue to give it a thought to the naming of these objects?

Reply:
KTensor (SyftTensor) is just a util class to help create IValue later, and keep the data and the shape separated

I accidentaly saw the namming issue on slack. Out of curiosity, from the PR, what is missing from it that keeps you from just using Tensor here? This would copy the data earlier but from the uses of KTensor in this PR it would be covered by Tensor (with the shape being in the tensor but available through .shape() AFAIU)
Maybe it would not be too hard to provide the necessary bindings.

Reply:
Make sense, I thought constructing the Tensor would be hard for the user at first, but it's not, and it's better not to introduce another data type here